src/pybind/mgr/cephadm/serve.py

```diff
@@ -633,9 +643,85 @@ def _apply_all_services(self) -> bool:
+                hosts_altered.update(daemon_deployed_hosts)
+            removed_daemons, removed_daemon_hosts = await self.remove_given_daemons(to_remove)
+            if removed_daemons:
+                r = True
+                self.mgr.spec_store.mark_needs_configuration(spec.service_name())
+                hosts_altered.update(conflict_hosts_altered)
+            if self.mgr.spec_store.needs_configuration(spec.service_name()):
+                svc = service_registry.get_service(spec.service_type)
+                svc.config(spec)
+                self.mgr.spec_store.mark_configured(spec.service_name())
+            return (r, hosts_altered, daemon_place_fails)
+
+        async def _deploy_and_remove_all(
+            all_conflicts: List[orchestrator.DaemonDescription],
+            all_to_deploy: List[CephadmDaemonDeploySpec],
+            all_to_remove: List[orchestrator.DaemonDescription]
+        ) -> List[Tuple[bool, Set[str], List[str]]]:
+            futures = []
+
+            for host in self.mgr.cache.get_hosts():
+                conflicting_daemons = [dd for dd in all_conflicts if dd.hostname == host]
+                daemons_to_deploy = [dd for dd in all_to_deploy if dd.host == host]
+                daemons_to_remove = [dd for dd in all_to_remove if dd.hostname == host]
+                futures.append(_parallel_deploy_and_remove(host, conflicting_daemons, daemons_to_deploy, daemons_to_remove))
+
+            return await gather(*futures)
+
+        deploy_names = [d.name() for d in all_daemons_to_deploy]
+        rm_names = [d.name() for d in all_conflicting_daemons] + [d.name() for d in all_daemons_to_remove]
+        with self.mgr.async_timeout_handler(cmd=f'cephadm deploying ({deploy_names} and removing {rm_names} daemons)'):
+            results = self.mgr.wait_async(_deploy_and_remove_all(all_conflicting_daemons, all_daemons_to_deploy, all_daemons_to_remove))
+
+        if any(res[0] for res in results):
+            changed = True
+
+        altered_hosts: Set[str] = set()
+        for altered_host in [res[1] for res in results]:
+            altered_hosts.update(altered_host)
+
+        placement_failures: List[str] = []
+        for place_failures in [res[2] for res in results]:
+            placement_failures.extend(place_failures)
+
+        if placement_failures:
+            self.mgr.set_health_warning('CEPHADM_DAEMON_PLACE_FAIL', f'Failed to place {len(placement_failures)} daemon(s)', len(
+                placement_failures), placement_failures)
+        if self.mgr.use_agent:
+            # can only send ack to agents if we know for sure port they bound to
+            altered_hosts = set([h for h in altered_hosts if (h in self.mgr.agent_cache.agent_ports and not self.mgr.cache.is_host_draining(h))])
+            self.mgr.agent_helpers._request_agent_acks(altered_hosts, increment=True)
+
+        self.mgr.daemon_deploy_queue.clear_queued_daemons()
+        self.mgr.daemon_removal_queue.clear_queued_daemons()
```

Comment:
I think John has already made a similar comment. We should remove the "queue" reference in the methods as it adds some repetition. We should just call `clear` in this case as from the variable name its clear what are we clearing.